src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -528,6 +529,20 @@ public T WithClientCapabilities(IEnumerable<string> clientCapabilities)
             return (T)this;
         }
 
+        /// <summary>
+        /// Determines whether or not instance discovery is performed when attempting to authenticate. Setting this to false will completely disable
+        /// instance discovery and authority validation. This will not affect the behavior of application configured with regional endpoints however.
+        /// Also, you can still provide your own instance metadata.
+        /// </summary>
+        /// <param name="enableInstanceDiscovery">Determines if instance discovery/Authority validation is performed</param>
+        /// <returns></returns>
+        public T WithInstanceDiscoevryEndpoint(bool enableInstanceDiscovery)
```

Comment:
Can this be renamed to `WithInstanceDiscoverDisabled(bool disableInstanceDiscovery = true)` or `WithDisabledInstanceDiscovery(bool disableInstanceDiscovery = true)`

Reply:
I think most of our APIs that take in a "bool" have it as `WithXYZ(enabled)`. 

But I believe that @rayluo 's design mentions `WithInstanceDiscovery` and not `WithInstanceDiscoveryEndpoint`. Naming in hard.

Reply:
The design mentions `InstanceDiscoveryEndpoint` but I think `WithInstanceDiscovery` is ok

Reply:
> The design mentions `InstanceDiscoveryEndpoint` but I think `WithInstanceDiscovery` is ok

I can explain. The design doc written as a WORD doc with table, was the first several iterations. At that time, the `InstanceDiscoveryEndpoint` was indeed proposed, because we wanted the polymorphism to accept not only a boolean but also a new string endpoint, and the latter was considered a major reason to "justify" this new parameter. However, that "customizing a new endpoint" idea was later dropped due to lack of real-world requirements. And we eventually ended up with a [simplified design](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=/AuthorityValidation/authority_validation_concepts_and_api.md&_a=preview&anchor=skip-authority-validation-and-alias-metadata-behaviors-without-a-set-of-predefined-authorities), which chose the name `WithInstanceDiscovery`.